File: templates/template.java
// run on java 1.8.0_162

import java.io.*;
import java.util.*;

public class Main {

	static class FastScanner {
		private BufferedReader br;
		private StringTokenizer st;

		FastScanner() {
			br = new BufferedReader(new InputStreamReader(System.in));
		}

		String next() {
			while (st == null || !st.hasMoreElements())
				try {
					st = new StringTokenizer(br.readLine());
				} catch (IOException e) {
					throw new IllegalStateException(e);
				}
			return st.nextToken();
		}

		int nextInt() {
			return Integer.parseInt(next());
		}

		long nextLong() {
			return Long.parseLong(next());
		}
	}

	public static void main(String args[]) {
		FastScanner cin = new FastScanner();
		PrintWriter cout = new PrintWriter(System.out);

		int n = cin.nextInt();
		String s = cin.next();
		int[] x = new int[n];
		int[] y = new int[n];
		for (int i = 0; i < n; i++) {
			x[i] = cin.nextInt();
			y[i] = cin.nextInt();
		}
		
		// solve here

		cout.println(n);
		cout.println(s);
		cout.println(x[0]);
		cout.println(y[n - 1]);

		cout.flush();
	}

	private static int solve(int n) {
	  return 0;
	}

}

File: templates/template.kt
// run on kotlin 1.3.10

import java.io.*
import java.util.*

class FastScanner(stream: InputStream) {
  private val reader: BufferedReader = BufferedReader(InputStreamReader(stream), 32768)
  private var tokenizer: StringTokenizer? = null
  init {
    tokenizer = null
  }
  operator fun next(): String {
    while (tokenizer == null || !tokenizer!!.hasMoreTokens()) {
      tokenizer = StringTokenizer(reader.readLine())
    }
    return tokenizer!!.nextToken()
  }
  fun nextLine(): String? {
    val fullLine: String
    while (tokenizer == null || !tokenizer!!.hasMoreTokens()) {
      fullLine = reader.readLine()
      return fullLine
    }
    return null
  }
  fun nextInt(): Int {
    return Integer.parseInt(next())
  }
  fun nextLong(): Long {
    return java.lang.Long.parseLong(next())
  }

}

fun main() {
	val cin = FastScanner(BufferedInputStream(System.`in`))
	val cout = PrintWriter(BufferedOutputStream(System.out))

	val n = cin.nextInt()
	val s = cin.next() // or nextLine()!!
	val x = IntArray(n)
	val y = IntArray(n)
	for (i in 0..n - 1) {
		x[i] = cin.nextInt()
		y[i] = cin.nextInt()
	}
	
	// solve here

	println(n)
	println(s)
	println(x[0])
	println(y[n-1])

	cout.close()
}

fun solve(n:Int):Int {
	return 0
}
